.github/workflows/create-release.yml

```
@@ -27,10 +27,47 @@ jobs:
           python-version: 3.8
       - name: Change version number
         id: version
+        # See this issue for explanation and testing:
+        # https://github.com/cmu-delphi/delphi-epidata/pull/1473
         run: |
           python -m pip install bump2version
-          echo -n "::set-output name=next_tag::"
-          bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,
+          allowed_pattern="^(major|minor|patch|[0-9]+\.[0-9]+\.[0-9]+)$"
+
+          if [[ ! ${{ github.event.inputs.versionName }} =~ $allowed_pattern ]]; then
+            echo "\nInvalid version name: ${{ github.event.inputs.versionName }}"
+            exit 1
+          fi
+
+          if [[ ${{ github.event.inputs.versionName }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            # use given version number
+            NEXT_TAG="${{ github.event.inputs.versionName }}"
+          else
+            # calculate new version number based on given tag
+            NEXT_TAG=$(bump2version --dry-run --list ${{ github.event.inputs.versionName }} | grep ^new_version | sed -r s,"^.*=",,)
+          fi
+          # apply given or calculated version number
+          bump2version --new-version $NEXT_TAG _ignored_arg_
+          # save version number for later
+          echo "next_tag=$NEXT_TAG" >> $GITHUB_OUTPUT
+      - name: Get main branch SHA
+        id: base-sha
+        run: echo "sha=$(git rev-parse origin/main)" >> $GITHUB_OUTPUT
+      - name: Get Python client file changes
+        id: changed-py
+        uses: dorny/paths-filter@v3
+        with:
+            base: "${{ steps.base-sha.outputs.sha }}"
```

does this work?  its more like the usage in https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/.github/workflows/create-release.yml#L32-L40, and more straightforward about its intent
```suggestion
      - name: Get Python client file changes
        id: changed-py
        uses: dorny/paths-filter@v3
        with:
            base: "main"
            ref: "dev"
```